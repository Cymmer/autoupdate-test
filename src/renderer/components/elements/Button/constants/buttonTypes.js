const buttonTypes = {
  PRIMARY: {
    BLUE: 'primary___blue',
    GREEN: 'primary___green',
    RED: 'primary___red',
    YELLOW: 'primary___yellow',
    PURPLE: 'primary___purple',
  },
  SECONDARY: {
    BLUE: 'secondary___blue',
    GREEN: 'secondary___green',
    RED: 'secondary___red',
    YELLOW: 'secondary___yellow',
    PURPLE: 'secondary___purple',
    WHITE: 'secondary___white',
  },
  TEXT: {
    BLUE: 'text___blue',
    GREEN: 'text___green',
    RED: 'text___red',
    YELLOW: 'text___yellow',
    PURPLE: 'text___purple',
    GRAY: 'text___gray',
  },
  TERTIARY: 'tertiary',
};

export default buttonTypes;
