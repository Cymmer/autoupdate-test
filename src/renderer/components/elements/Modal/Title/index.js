import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
import GLOBALS from 'codechum-app-globals';
import styles from './styles.module.scss';

import Icon from '../../Icon';
import Text from '../../Text';
import textTypes from '../../Text/constants/textTypes';

import modalTitleTypes from '../constants/modalTitleTypes';

const ModalTitle = ({
  id,
  iconId,
  type,
  icon,
  className,
  title,
  subject,
  info,
  titleColorClass,
  textClassName,
  iconClassName,
}) => (
  <div className={cn(className, styles.ModalTitle)}>
    <Icon
      className={cn(iconClassName, styles[`ModalTitle___${type}_icon`])}
      icon={icon}
      id={iconId}
    />
    <div className={styles.ModalTitle_info}>
      <div className={styles.ModalTitle_title}>
        {type === modalTitleTypes.SMALL && (
          <Text
            className={textClassName}
            colorClass={titleColorClass || GLOBALS.COLOR_CLASSES.BLUE['300']}
            id={id}
            type={textTypes.HEADING.XXS}
          >
            {title}
          </Text>
        )}
        {type === modalTitleTypes.LARGE && (
          <>
            <Text
              className={styles.ModalTitle___large_title}
              colorClass={GLOBALS.COLOR_CLASSES.NEUTRAL['400']}
              id={id}
              type={textTypes.BODY.MD}
            >
              {title}
            </Text>
            <Text
              colorClass={
                titleColorClass || GLOBALS.COLOR_CLASSES.NEUTRAL['700']
              }
              type={textTypes.HEADING.SM}
            >
              {subject}
            </Text>
          </>
        )}
      </div>
      {type === modalTitleTypes.LARGE && (
        <Text
          className={styles.ModalTitle___large_info}
          colorClass={GLOBALS.COLOR_CLASSES.NEUTRAL['400']}
          type={textTypes.BODY.SM}
        >
          {info}
        </Text>
      )}
    </div>
  </div>
);

ModalTitle.defaultProps = {
  id: null,
  iconId: null,
  type: modalTitleTypes.LARGE,
  className: null,
  subject: null,
  info: null,
  titleColorClass: null,
  iconClassName: null,
  textClassName: null,
};

ModalTitle.propTypes = {
  iconId: PropTypes.string,
  id: PropTypes.string,
  type: PropTypes.oneOf([modalTitleTypes.LARGE, modalTitleTypes.SMALL]),
  icon: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  className: PropTypes.string,
  subject: PropTypes.string,
  info: PropTypes.string,
  titleColorClass: PropTypes.string,
  iconClassName: PropTypes.string,
  textClassName: PropTypes.string,
};

export default ModalTitle;
