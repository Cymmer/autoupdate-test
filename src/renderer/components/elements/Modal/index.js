import React from 'react';
import cn from 'classnames';
import PropTypes from 'prop-types';
import ReactModal from 'react-modal';
import GLOBALS from 'codechum-app-globals';
import styles from './styles.module.scss';

import Icon from '../Icon';
import Spinner from '../Spinner';
import ButtonGroup from '../Button/Group';
import buttonTypes from '../Button/constants/buttonTypes';

import modalSizes from './constants/modalSizes';
import modalPositions from './constants/modalPositions';

const Modal = ({
  actions,
  children,
  className,
  closeModalId,
  handleClose,
  hasCloseButton,
  id,
  isLoading,
  isOpen,
  overlayClassName,
  parentSelector,
  position,
  size,
}) => (
  <ReactModal
    ariaHideApp
    shouldCloseOnEsc
    shouldCloseOnOverlayClick
    shouldFocusAfterRender
    // second condition is just a temporary workaround for tests
    shouldReturnFocusAfterClose
    bodyOpenClassName={styles.Modal_body___open}
    className={cn(
      className,
      styles[`Modal___${size}`],
      styles[`Modal___${position}`]
    )}
    contentLabel="Modal"
    htmlOpenClassName={styles.Modal_html___open}
    id={id}
    isOpen={isOpen}
    overlayClassName={cn(
      styles[`Modal_overlay___${position}`],
      overlayClassName
    )}
    parentSelector={
      parentSelector && process.env.NODE_ENV !== 'test'
        ? () => document.querySelector(parentSelector)
        : () => document.querySelector('body')
    }
    portalClassName={styles.Modal_portal}
    onRequestClose={!isLoading ? handleClose : null}
  >
    {isLoading && (
      <>
        <Spinner className={styles.Modal___loading_spinner} />
        <div className={styles.Modal___loading_background} />
      </>
    )}
    {hasCloseButton && (
      <button
        className={styles.Modal_close}
        data-test="closeButton"
        id={closeModalId}
        type="button"
        onClick={handleClose}
      >
        <Icon className={styles.Modal_close_icon} icon="close" />
      </button>
    )}
    {children}
    {actions && <ButtonGroup buttons={actions} />}
  </ReactModal>
);

ReactModal.setAppElement(document.getElementById('root'));

Modal.defaultProps = {
  className: null,
  closeModalId: null,
  id: null,
  size: modalSizes.MD,
  position: modalPositions.TOP,
  handleClose: null,
  hasCloseButton: true,
  actions: null,
  isLoading: false,
  parentSelector: null,
};

Modal.propTypes = {
  className: PropTypes.string,
  closeModalId: PropTypes.string,
  id: PropTypes.string,
  isLoading: PropTypes.bool,
  size: PropTypes.oneOf([
    modalSizes.LG,
    modalSizes.MD,
    modalSizes.SM,
    modalSizes.XS,
  ]),
  position: PropTypes.oneOf([modalPositions.CENTER, modalPositions.TOP]),
  children: PropTypes.node.isRequired,
  handleClose: PropTypes.func,
  hasCloseButton: PropTypes.bool,
  isOpen: PropTypes.bool.isRequired,
  overlayClassName: PropTypes.string,

  // for mapping the buttons at the bottom of the modal
  actions: PropTypes.arrayOf(
    PropTypes.shape({
      text: PropTypes.oneOfType([PropTypes.string, PropTypes.element])
        .isRequired,
      type: PropTypes.oneOf([
        buttonTypes.PRIMARY.BLUE,
        buttonTypes.PRIMARY.RED,
        buttonTypes.PRIMARY.GREEN,
        buttonTypes.PRIMARY.YELLOW,
        buttonTypes.SECONDARY.BLUE,
        buttonTypes.SECONDARY.GREEN,
        buttonTypes.SECONDARY.RED,
        buttonTypes.SECONDARY.YELLOW,
        buttonTypes.TEXT.BLUE,
        buttonTypes.TEXT.RED,
        buttonTypes.TEXT.GREEN,
        buttonTypes.TERTIARY,
      ]),
      kind: PropTypes.oneOf([
        GLOBALS.BUTTON_KINDS.BUTTON,
        GLOBALS.BUTTON_KINDS.SUBMIT,
        GLOBALS.BUTTON_KINDS.RESET,
      ]),
      disabled: PropTypes.bool,
      onClick: PropTypes.func.isRequired,
      isLoading: PropTypes.bool,
      dataTestId: PropTypes.string,
    })
  ),

  parentSelector: PropTypes.string,
};

export default Modal;
