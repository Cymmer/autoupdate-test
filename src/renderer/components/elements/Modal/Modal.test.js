import React from 'react';
import { shallow, mount } from 'enzyme';
import { findByTestAttr } from 'codechum-app-utils';
import GLOBALS from 'codechum-app-globals';
import Modal from './index';
import buttonTypes from '../Button/constants/buttonTypes';

describe('RENDERING - Modal', () => {
  it('should render with default props', () => {
    const component = shallow(<Modal isOpen>Test</Modal>);
    expect(component).toMatchSnapshot();
  });

  it('should render the Spinner if it is loading', () => {
    const component = shallow(
      <Modal isLoading isOpen>
        Test
      </Modal>
    );
    expect(component).toMatchSnapshot();
  });

  it('should render the contents if it is not loading', () => {
    const component = shallow(
      <Modal isOpen isLoading={false}>
        Test
      </Modal>
    );
    expect(component).toMatchSnapshot();
  });

  it('should render close button if hasCloseButton is true', () => {
    const component = shallow(
      <Modal hasCloseButton isOpen isLoading={false}>
        Test
      </Modal>
    );
    expect(component).toMatchSnapshot();
  });

  it('should render the actions if it has actions', () => {
    const component = shallow(
      <Modal
        isOpen
        actions={[
          {
            text: 'Action 1',
            type: buttonTypes.PRIMARY.BLUE,
            kind: GLOBALS.BUTTON_KINDS.BUTTON,
            disabled: false,
            onClick: () => {},
          },
          {
            text: 'Action 2',
            type: buttonTypes.PRIMARY.GREEN,
            kind: GLOBALS.BUTTON_KINDS.BUTTON,
            onClick: () => {},
          },
        ]}
        isLoading={false}
      >
        Test
      </Modal>
    );
    expect(component).toMatchSnapshot();
  });
});

describe('EVENTS - Modal', () => {
  it('should show a clickable button if it hasCloseButton', () => {
    const clickFn = jest.fn();
    const component = mount(
      <Modal hasCloseButton isOpen handleClose={clickFn} isLoading={false}>
        Test
      </Modal>
    );

    const closeButton = findByTestAttr(component, 'closeButton');
    closeButton.simulate('click');

    expect(clickFn).toHaveBeenCalled();
  });
});
