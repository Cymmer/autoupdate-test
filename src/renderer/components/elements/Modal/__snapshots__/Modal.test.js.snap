// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RENDERING - Modal should render close button if hasCloseButton is true 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="Modal_body___open"
  className="Modal___md Modal___top"
  closeTimeoutMS={0}
  contentElement={[Function]}
  contentLabel="Modal"
  htmlOpenClassName="Modal_html___open"
  id={null}
  isOpen={true}
  onRequestClose={null}
  overlayClassName="Modal_overlay___top"
  overlayElement={[Function]}
  parentSelector={[Function]}
  portalClassName="Modal_portal"
  preventScroll={false}
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <button
    className="Modal_close"
    data-test="closeButton"
    id={null}
    onClick={null}
    type="button"
  >
    <Icon
      className="Modal_close_icon"
      icon="close"
      id={null}
      isPulsating={false}
      style={null}
    />
  </button>
  Test
</Modal>
`;

exports[`RENDERING - Modal should render the Spinner if it is loading 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="Modal_body___open"
  className="Modal___md Modal___top"
  closeTimeoutMS={0}
  contentElement={[Function]}
  contentLabel="Modal"
  htmlOpenClassName="Modal_html___open"
  id={null}
  isOpen={true}
  onRequestClose={null}
  overlayClassName="Modal_overlay___top"
  overlayElement={[Function]}
  parentSelector={[Function]}
  portalClassName="Modal_portal"
  preventScroll={false}
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <Spinner
    className="Modal___loading_spinner"
    colorName="blue"
    size="lg"
  />
  <div
    className="Modal___loading_background"
  />
  <button
    className="Modal_close"
    data-test="closeButton"
    id={null}
    onClick={null}
    type="button"
  >
    <Icon
      className="Modal_close_icon"
      icon="close"
      id={null}
      isPulsating={false}
      style={null}
    />
  </button>
  Test
</Modal>
`;

exports[`RENDERING - Modal should render the actions if it has actions 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="Modal_body___open"
  className="Modal___md Modal___top"
  closeTimeoutMS={0}
  contentElement={[Function]}
  contentLabel="Modal"
  htmlOpenClassName="Modal_html___open"
  id={null}
  isOpen={true}
  onRequestClose={null}
  overlayClassName="Modal_overlay___top"
  overlayElement={[Function]}
  parentSelector={[Function]}
  portalClassName="Modal_portal"
  preventScroll={false}
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <button
    className="Modal_close"
    data-test="closeButton"
    id={null}
    onClick={null}
    type="button"
  >
    <Icon
      className="Modal_close_icon"
      icon="close"
      id={null}
      isPulsating={false}
      style={null}
    />
  </button>
  Test
  <ButtonGroup
    buttons={
      Array [
        Object {
          "disabled": false,
          "kind": "button",
          "onClick": [Function],
          "text": "Action 1",
          "type": "primary___blue",
        },
        Object {
          "kind": "button",
          "onClick": [Function],
          "text": "Action 2",
          "type": "primary___green",
        },
      ]
    }
    className={null}
    direction="horizontal"
  />
</Modal>
`;

exports[`RENDERING - Modal should render the contents if it is not loading 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="Modal_body___open"
  className="Modal___md Modal___top"
  closeTimeoutMS={0}
  contentElement={[Function]}
  contentLabel="Modal"
  htmlOpenClassName="Modal_html___open"
  id={null}
  isOpen={true}
  onRequestClose={null}
  overlayClassName="Modal_overlay___top"
  overlayElement={[Function]}
  parentSelector={[Function]}
  portalClassName="Modal_portal"
  preventScroll={false}
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <button
    className="Modal_close"
    data-test="closeButton"
    id={null}
    onClick={null}
    type="button"
  >
    <Icon
      className="Modal_close_icon"
      icon="close"
      id={null}
      isPulsating={false}
      style={null}
    />
  </button>
  Test
</Modal>
`;

exports[`RENDERING - Modal should render with default props 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="Modal_body___open"
  className="Modal___md Modal___top"
  closeTimeoutMS={0}
  contentElement={[Function]}
  contentLabel="Modal"
  htmlOpenClassName="Modal_html___open"
  id={null}
  isOpen={true}
  onRequestClose={null}
  overlayClassName="Modal_overlay___top"
  overlayElement={[Function]}
  parentSelector={[Function]}
  portalClassName="Modal_portal"
  preventScroll={false}
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <button
    className="Modal_close"
    data-test="closeButton"
    id={null}
    onClick={null}
    type="button"
  >
    <Icon
      className="Modal_close_icon"
      icon="close"
      id={null}
      isPulsating={false}
      style={null}
    />
  </button>
  Test
</Modal>
`;
